src/lib/erc20.ts

```diff
@@ -0,0 +1,33 @@
+import { ThirdwebContract, toTokens } from "thirdweb";
+import { getActiveClaimCondition } from "thirdweb/extensions/erc20";
+import { getContractMetadata } from "thirdweb/extensions/common";
+import { fetchCurrencyMetadata } from "@/lib/utils";
+
+export async function getERC20Info(contract: ThirdwebContract) {
+    try {
+        const claimCondition = await getActiveClaimCondition({ contract });
+        const contractMetadata = await getContractMetadata({ contract });
+        const priceInWei = claimCondition?.pricePerToken;
+        const currencyMetadata = claimCondition?.currency
+            ? await fetchCurrencyMetadata(claimCondition.currency).catch((err) => {
+                console.error("Error fetching currency metadata:", err);
+                throw err

```

Comment:
same here, no try catch, no extra logging. let it throw. the error you log at the top level will already tell you where its failing